### DISCUSSION TITLE
[V3] Modelable Attribute not working with array/form object

### DISCUSSION THREAD

**Discussion by quentinhnrt** [livewire/livewire#6106] [7. Bugs (archived)]
### Livewire version

v3.0.0-beta.5

### Which PHP version are you using?

PHP 8.2

### Steps To Reproduce

1. Have a livewire component with a array public property
```php
<?php

namespace App\Livewire;

use Livewire\Component;

class ParentComponent extends Component
{
    public array $data = [
        'foo' => 0
   ];
    public function render()
    {
        return view('livewire.parent-component');
    }
}
```
2. Have a child component with a Modelable property
```php
<?php

namespace App\Livewire;

use Livewire\Attributes\Modelable;
use Livewire\Component;

class ChildComponent extends Component
{
    #[Modelable]
    public int $value = 0;
    public function render()
    {
        return view('livewire.child-component');
    }
}

```
3. Add a wire:model on the child component targeting a index of the array property (foo.bar)

```html
<div>
    <livewire:child-component wire:model="data.foo" />
</div>
```

=> Result
`Property [$data.foo] not found on component: [parent-component]`

I've actually found a workaround, the actual Modelable attribute don't take in count if we are targeting an array (or form object), so in order to get the value of the targeted property i used the data_get() method in the Modelable attribute like this :

=>

```php
<?php

namespace App\Livewire\Attributes;

use Livewire\Features\SupportAttributes\Attribute as LivewireAttribute;
use function Livewire\store;

#[\Attribute]
class CustomModelable extends LivewireAttribute
{
    public function mount($params, $parent)
    {
        if ($parent && isset($params['wire:model'])) {
            $outer = $params['wire:model'];
            $inner = $this->getName();

            store($this->component)->push('bindings', $inner, $outer);
            
            $value = data_get($parent, $outer);

            $this->setValue($value);
        }
    }
}
```

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by joshhanley** [comment]
@quentinhnrt thanks for reporting! Any chance you could submit a failing test PR so we can investigate? Thanks!

**Reply by quentinhnrt** [reply]
No problem i'm on it !

**Comment by vincentdev15** [comment]
I have the same problem.

**Comment by rinodrummer** [comment]
Similar issue here, but I have an array inside an array:

```php
$form = [
    'entries' => [],
];
```

**Comment by djfrailey** [comment]
I am also running into the same problem.

**Reply by djfrailey** [reply]
I am removing my upvote from this as I figured out what was going on.

I'm working on an order form, below is a very minimal reproduction of what I have going on:

An OrderForm Livewire component that is a Full Page component.
```
class OrderForm extends Component
{
    public array $order = [];

    public function addItem()
    {
        $this->order[] = [
            'partNumber' => '',
            'quantity' => 
        ];
    }

    public function duplicateItem($index)
    {
        $this->order[] = $this->order[$index];
    }
}
```
```
<form wire:submit="save">
    <!-- Top Level Order Form Fields -->
    @foreach($order as $index => $item)
        <x-order-item :$item :$index />
    @endforeach
</form>
```

An OrderItem Blade Component that renders each individual Order Item.
```
<div class="order-item">
    <div>
        <livewire:part-number-autocomplete
            label="Part Number"
            wire:model="currentOrder.{{$index}}.partNumber"
        />
        <x-input
            label="Quantity"
            wire:model="quantity"
        />
    </div>
    <div>
        <button wire:click="duplicateItem({{$index}})">Duplicate</button>
        <button wire:click="addItem()">Add Item</button>
    </div>
</div>
```

An autocomplete Livewire component
```
class PartNumberAutocomplete extends Component
{
    #[Modelable]
    public ?string $value = null;

    // search function and whatnot
}
```

I was getting the exact same error as everyone else

`Property [$currentOrder.0.partNumber] not found on component: [order-form]`

Then I added `wire:key` to the child Livewire component and it seems to be working fine.
